### ISSUE TITLE
Tag Categories MySQL Error

### ISSUE THREAD

**Issue by PhuriousGeorge** [manyfold3d/manyfold#2735]
**Describe the bug**
Once upon a time I was able to upload models and apply tags with `character: zelda` or `series: star wars` in order to group tags under a common heading instead of making a big mess of tags.

Today I created a new instance using MaraDB for the database and linuxserver.io's image (Manyfold v0.79.0) and any time I attempt to add a tag similar to the above, the "Models" page refuses to load with "We're sorry, but something went wrong. If you are the application owner check the logs for more information."  As soon as I remove the colon-separated tag, the "Models" page is once again accessible.

Tested and confirmed with both Manyfold and linuxserver.io images.  Appears to be an issue with mysql2 implementation.  From the log:
```15:23:59 rails.1              | F, [2024-09-21T15:23:59.988230 #197] FATAL -- : [e47473a7-3cfd-4909-a156-6414db7d0f76]   
15:23:59 rails.1              | [e47473a7-3cfd-4909-a156-6414db7d0f76] ActionView::Template::Error (Mysql2::Error: You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near '~ '^series:' and taggings.taggable_id=models.id and taggings.taggable_type='M...' at line 1):
15:23:59 rails.1              | [e47473a7-3cfd-4909-a156-6414db7d0f76]     4:        tierunset = tiers.map { |tier|
15:23:59 rails.1              | [e47473a7-3cfd-4909-a156-6414db7d0f76]     5:          reg = ActiveRecord::Base.connection.quote("^" + tier + ":")
15:23:59 rails.1              | [e47473a7-3cfd-4909-a156-6414db7d0f76]     6:          regact = Rails.env.development? ? "REGEXP" : "~"
15:23:59 rails.1              | [e47473a7-3cfd-4909-a156-6414db7d0f76]     7:          [tier, @models.where("(select count(*) from tags join taggings on tags.id=taggings.tag_id where tags.name #{regact} #{reg} and taggings.taggable_id=models.id and taggings.taggable_type='Model')<1").count]
15:23:59 rails.1              | [e47473a7-3cfd-4909-a156-6414db7d0f76]     8:        }.to_h
15:23:59 rails.1              | [e47473a7-3cfd-4909-a156-6414db7d0f76]     9:      end %>
15:23:59 rails.1              | [e47473a7-3cfd-4909-a156-6414db7d0f76]    10: <% end %>
15:23:59 rails.1              | [e47473a7-3cfd-4909-a156-6414db7d0f76]
```

**To Reproduce**
1. Edit existing model
2. Add `series: star wars`
3. Save
4. Click on "Models"
5. Receive error

**Expected behavior**
I would expect the application to behave how it does under my previous setup using PostgreSQL and manyfold's image.

**Desktop (please complete the following information):**
 - OS: Docker
 - Browser Chrome
 - Version 0.79.0 (linuxserver.io)

**Comment by Floppy** [comment]
Thanks - that certainly should still work, I'll check it out!

**Comment by Floppy** [comment]
Ah, I see, this is to do with the regexp syntax in MySQL.

**Comment by Floppy** [comment]
Aha - there are two places in the code that check which regexp syntax should be used. One of them was wrong. I've fixed it, and it should be released soon in v0.80.0. Thanks for the report!

**Comment by PhuriousGeorge** [comment]
Awesome, glad you caught it! I was poking around, but regex is the bane of my existence
